udata_front/theme/gouvfr/templates/dataset/display.html

```diff
@@ -39,7 +39,7 @@
 {% endblock %}
 
 {% block content %}
-{% cache cache_duration, 'dataset-content', dataset.id|string, reuses.page|string, g.lang_code, current_user.slug or 'anonymous' %}
+{% cache cache_duration, 'dataset-content', dataset.id|string, g.lang_code, current_user.slug or 'anonymous' %}
```

Comment:
I guess `reuses.page|string` was invalidating the cache when a reuse or added or deleted to the dataset, isn't interesting to keep?

Reply:
Oooh I thought the same, but it's actually the reuse page number. **It is indeed necessary for pagination**.

The same key is probably needed on reuse and orga pages to paginate the different elements.

Reply:
The reuses.page is not know in the event context. We cannot recreate such a cache key in the signal.
I do not understand your comment @maudetes  :/

Reply:
> The reuses.page is not know in the event context. We cannot recreate such a cache key in the signal.

Maybe using the modified date as @abulte suggested would do it. We wouldn't need to manually invalidate the cache here.
If it somehow doesn't work, we would need to iterate on all reuses pages (as you do with lang) to invalidate the cache for all reuses pages.

> I do not understand your comment @maudetes :/

I'm talking about the paginating inside a dataset, reuse or orga page, such as https://www.data.gouv.fr/fr/datasets/base-sirene-des-entreprises-et-de-leurs-etablissements-siren-siret/?reuses_page=2#community-reuses. If you don't use the current page number in the key, you would be unable to paginate. The previous link would return the result for the `reuses_page=1`